test/controllers/ConfirmTransferControllerSpec.scala

```diff
@@ -76,6 +78,102 @@ class ConfirmTransferControllerSpec extends FrontEndTestHelper {
       val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
       mockGraphqlConsignmentSummaryResponse(dataString)
 
+      val confirmTransferPage = controller
+        .confirmTransfer(consignmentId)
+        .apply(FakeRequest(GET, s"/consignment/$consignmentId/confirm-transfer").withCSRFToken)
+
+      playStatus(confirmTransferPage) mustBe SEE_OTHER
+      redirectLocation(confirmTransferPage).get must equal(s"/consignment/$consignmentId/series")
+    }
+
+    "redirect to the transfer agreement page if the transfer agreement for that consignment has not been signed or agreed to" in {
+      val consignmentId = UUID.randomUUID()
+
+      val client = new GraphQLConfiguration(app.configuration).getClient[gcs.Data, gcs.Variables]()
+      val controller = instantiateConfirmTransferController(getAuthorisedSecurityComponents)
+      val consignmentStatuses = List(
+        ConsignmentStatuses(UUID.randomUUID(), UUID.randomUUID(), "Series", "Completed", someDateTime, None)
+      )
+      setConsignmentStatusResponse(app.configuration, wiremockServer, consignmentStatuses = consignmentStatuses)
+
+      val consignmentSummaryResponse: gcs.GetConsignment = getConsignmentSummaryResponse
+      val data: client.GraphqlData = client.GraphqlData(Some(gcs.Data(Some(consignmentSummaryResponse))), List())
+      val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
+      mockGraphqlConsignmentSummaryResponse(dataString)
+
+      val confirmTransferPage = controller
+        .confirmTransfer(consignmentId)
+        .apply(FakeRequest(GET, s"/consignment/$consignmentId/confirm-transfer").withCSRFToken)
+
+      playStatus(confirmTransferPage) mustBe SEE_OTHER
+      redirectLocation(confirmTransferPage).get must equal(s"/consignment/$consignmentId/transfer-agreement")
+    }
+
+    "redirect to the upload page if the upload status is not 'Completed'" in {
```

Comment:
Maybe change the test title slightly to: `... upload page with an authenticated user if the upload ...`